lib/mail/utilities.rb

```diff
@@ -72,12 +72,25 @@ def dquote( str )
     #  unqoute(string) #=> 'This is "a string"'
     def unquote( str )
       if str =~ /^"(.*?)"$/
-        $1.gsub(/\\(.)/, '\1')
+        unescape($1)
       else
         str
       end
     end
 
+    # Removes any \-escaping.
+    #
+    # Example:
+    #
+    #  string = 'This is \"a string\"'
+    #  unescape(string) #=> 'This is "a string"'
+    #
+    #  string = '"This is \"a string\""'
+    #  unescape(string) #=> '"This is "a string""'
+    def unescape( str )
+      str.gsub(/\\(.)/, '\1')
```

Comment:
Looking at the RFC https://tools.ietf.org/html/rfc2822#section-3.2.2 seems that it would be safe to do `/\\"/` and only replace quoted pairs, as `\` is a valid char within the text. This might not catch double escaping like `\\"` but that is sort of expected.